sunpy/physics/differential_rotation.py

```diff
@@ -84,138 +89,315 @@ def diff_rot(duration: u.s, latitude: u.deg, rot_type='howard', frame_time='side
+        pixel indices, and values (:, 1) are the x values of the pixel indices.
+
+    observer : `~astropy.coordinates.SkyCoord`
+        The location of the observer.
+
+    diff_rot_kwargs : None, `~dict`
+        Keyword arguments accepted by `sunpy.physics.differential_rotation.diff_rot`.
+
+    Returns
+    -------
+    coordinates : `~astropy.coordinates.SkyCoord`
+        The coordinates of a rotated edge.
+    """
+    # Coordinates
+    c = smap.pixel_to_world(pixels[:, 1], pixels[:, 0])
+
+    # Only apply solar rotation if all coordinates are on the disk.
+    if np.all(~coordinate_is_on_disk(c, smap.rsun_obs)):
+        coordinates = deepcopy(c)
+    else:
+        coordinates = solar_rotate_coordinate(c, observer=observer, **diff_rot_kwargs)
+    return coordinates
+
+
+def _get_extreme_position(coords, axis, operator=np.nanmax):
+    """
+    Helper function that calculates an extreme position from a list of
+    coordinates.
+
+    Parameters
+    ----------
+    coords : `~list`
+        Each member of the list is a `~astropy.coordinates.SkyCoord`.
+
+    axis :  'Tx', 'Ty'
+        Which helioprojective axis to examine.
+
+    operator : numpy function
+        A numpy function that finds an extreme value in an array
+        of helioprojective coordinate values.
+
+    Returns
+    -------
+    extreme value : `~astropy.units.Quantity`
+        An extreme position in units of arcseconds.
+    """
+
+    extreme_values = []
+    for coord in coords:
+        if axis == 'Tx':
+            extreme_value = operator(coord.Tx.value)
+        elif axis == 'Ty':
+            extreme_value = operator(coord.Ty.value)
+        else:
+            raise ValueError('The "axis" argument must be either "Tx" or "Ty".')
+        extreme_values.append(extreme_value)
+
+    return operator(extreme_values) * u.arcsec
+
+
+def _get_bounding_coordinates(coords):
     """
-    Function that returns a new list of coordinates for each input coord.
+    Helper function that returns the bottom left and top right coordinates
+    that define a bounding box enclosing the passed in coordinates.
+
+    Parameters
+    ----------
+    coords : `~list`
+        Each member of the list is a `~astropy.coordinates.SkyCoord`.
+
+    Returns
+    -------
+    bottom_left, top_right : `~list`
+        A pair of `~astropy.coordinates.SkyCoord` that specify the
+        bottom left hand and top right hand corner of a bounding box that
+        minimally encloses all the input coordinates.
+    """
+    rotated_x_min = _get_extreme_position(coords, "Tx", operator=np.nanmin)
+    rotated_x_max = _get_extreme_position(coords, "Tx", operator=np.nanmax)
+    rotated_y_min = _get_extreme_position(coords, "Ty", operator=np.nanmin)
+    rotated_y_max = _get_extreme_position(coords, "Ty", operator=np.nanmax)
+    return SkyCoord(rotated_x_min, rotated_y_min, frame=Helioprojective, observer=coords[0].observer),\
```

Comment:
Is there a good reason to return two SkyCoord objects here rather than a length 2 SkyCoord? If there is, it would be better to line break by using a tuple i.e. `()` rather than `\`?

This looks like it works btw:
```python
a, b = SkyCoord([0*u.deg,1*u.deg], [0*u.deg, 1*u.deg], observer='venus', obstime="2019-01-01", frame=HeliographicStonyhurst)
```